onnxruntime/core/providers/cpu/tensor/concat.cc

```diff
@@ -20,112 +21,146 @@ ONNX_CPU_OPERATOR_KERNEL(
     KernelDefBuilder().TypeConstraint("T", DataTypeImpl::AllTensorTypes()),
     Concat);
 
-Status ConcatBase::PrepareForCompute(OpKernelContext* ctx, int input_count, Prepare& p) const {
+// this method will be shared between 'Concat' and 'ConcatFromSequence' ('concat' and 'stack' modes) to validate inputs
+static Status PrepareForCompute(OpKernelContext* ctx, const std::vector<const Tensor*>& input_tensors,
+                                int64_t axis, bool is_stack, Prepare& p) {
+  int input_count = static_cast<int>(input_tensors.size());
+  // Must have atleast one input to concat
   ORT_RETURN_IF_NOT(input_count >= 1, "Must have 1 or more inputs");
-  const Tensor* tensor_pointer = ctx->Input<Tensor>(0);
-  if (tensor_pointer == nullptr) return Status(common::ONNXRUNTIME, common::FAIL, "input count mismatch");
+
+  const Tensor* tensor_pointer = input_tensors[0];
+  if (tensor_pointer == nullptr)
```

Comment:
This is not possible since the TensorSeq data structure contains a vector<Tensor> and as long as this vector has a size of non-zero, this ptr can't be NULL. I suggest we do the cardinality check in Compute() using input_tensors.empty()

Reply:
Sure, we could remove this check (but many ops have very similar checks and I think this got carried over from the previous implementation)

Reply:
Synced offline - we will do this in a consolidated manner for all ops later.